### PR TITLE
Finalise filter sequence diagram

### PR DESCRIPTION
**Pull request by jyztintan**

### CODE REVIEW

File: docs/diagrams/filter/FilterSequenceDiagram.puml [hunk 1]
```diff
@@ -76,16 +76,16 @@ PredicateProducer --> FilterCommandParser : predicate
 deactivate PredicateProducer
 
 create FilterCommand
-FilterCommandParser -> FilterCommand : new(predicate)
+FilterCommandParser -> FilterCommand :
 activate FilterCommand
 
-FilterCommand --> FilterCommandParser : c
+FilterCommand --> FilterCommandParser : c::FilterCommand
```
Comment (appleraincoat) [inline docs/diagrams/filter/FilterSequenceDiagram.puml:82]:
```suggestion
FilterCommand --> FilterCommandParser : c
```

File: docs/diagrams/filter/FilterSequenceDiagram.puml [hunk 2]
```diff
@@ -76,16 +76,16 @@ PredicateProducer --> FilterCommandParser : predicate
...
 deactivate FilterCommand
 
-FilterCommandParser --> RealodexParser : c
+FilterCommandParser --> RealodexParser : c::FilterCommand
```
Comment (appleraincoat) [inline docs/diagrams/filter/FilterSequenceDiagram.puml:85]:
```suggestion
FilterCommandParser --> RealodexParser : c
```

File: docs/diagrams/filter/FilterSequenceDiagram.puml [hunk 3]
```diff
@@ -76,16 +76,16 @@ PredicateProducer --> FilterCommandParser : predicate
...
 deactivate FilterCommandParser
 
-RealodexParser --> LogicManager : c
+RealodexParser --> LogicManager : c::FilterCommand
```
Comment (appleraincoat) [inline docs/diagrams/filter/FilterSequenceDiagram.puml:88]:
```suggestion
RealodexParser --> LogicManager : c
```

File: docs/diagrams/filter/FilterSequenceDiagram.puml [hunk 4]
```diff
@@ -101,10 +101,10 @@ create CommandResult
 FilterCommand -> CommandResult :
 activate CommandResult
 
-CommandResult --> FilterCommand
+CommandResult --> FilterCommand : r::CommandResult
```
Comment (appleraincoat) [inline docs/diagrams/filter/FilterSequenceDiagram.puml:104]:
```suggestion
CommandResult --> FilterCommand : r
```

File: docs/diagrams/filter/FilterSequenceDiagram.puml [hunk 5]
```diff
@@ -101,10 +101,10 @@ create CommandResult
...
 deactivate CommandResult
 
-FilterCommand --> LogicManager : result
+FilterCommand --> LogicManager : r::CommandResult
```
Comment (appleraincoat) [inline docs/diagrams/filter/FilterSequenceDiagram.puml:107]:
```suggestion
FilterCommand --> LogicManager : r
```

**Review by appleraincoat — APPROVED** [review]
Minor changes.
Take note to change line 8 and 9 to
participant "c:FilterCommand" as FilterCommand LOGIC_COLOR
participant "r:CommandResult" as CommandResult LOGIC_COLOR

**Review by UdhayaShan1 — APPROVED** [review]
LGTM

**Pull request merged**